src/back/wgsl/writer.rs

```diff
@@ -673,8 +672,96 @@ impl<W: Write> Writer<W> {
                 self.write_expr(module, value, func_ctx)?;
                 writeln!(self.out, ");")?;
             }
-            _ => {
-                return Err(Error::Unimplemented(format!("write_stmt {:?}", stmt)));
+            // TODO: copy-paste from glsl-out
+            Statement::Block(ref block) => {
+                write!(self.out, "{}", INDENT.repeat(indent))?;
+                writeln!(self.out, "{{")?;
+                for sta in block.iter() {
+                    // Increase the indentation to help with readability
+                    self.write_stmt(module, sta, func_ctx, indent + 1)?
+                }
+                writeln!(self.out, "{}}}", INDENT.repeat(indent))?
+            }
+            Statement::Switch {
+                selector,
+                ref cases,
+                ref default,
+            } => {
+                // Start the switch
+                write!(self.out, "{}", INDENT.repeat(indent))?;
+                write!(self.out, "switch(")?;
+                self.write_expr(module, selector, func_ctx)?;
+                writeln!(self.out, ") {{")?;
+
+                // Write all cases
+                let mut write_case = true;
+                if !cases.is_empty() {
```

Comment:
this is strange, we shouldn't have this condition

Reply:
I remembered why we need such a strange condition because wgsl frontend allows writing switch without cases: 
```wgsl
fn main() {
    var pos: f32;
    switch (3) {
        default: { pos = 3.0; }
    }
}
```